summariser/src/analyze.rs

```diff
@@ -269,6 +269,59 @@ pub(crate) fn gauge_stats(frame: DataFrame, column: &str) -> anyhow::Result<Gaug
     })
 }
 
+/// Get the [`GaugeStats`] for a frame and the given column, partitioned
+/// by tags.
```

Comment:
To be able to show stats for open connections partitioned by arc, this new function is required. Alternatively, zero arc and full arc connections could be recorded as separate metrics altogether.